akka-http-core/src/main/scala/akka/http/javadsl/model/ws/UpgradeToWebSocket.scala

```diff
@@ -36,6 +39,29 @@ trait UpgradeToWebSocket extends sm.HttpHeader {
   def handleMessagesWith(handlerFlow: Graph[FlowShape[Message, Message], _ <: Any], subprotocol: String): HttpResponse
 
   /**
+   * Convenience API for (safely, including Streamed ones) ignoring all incoming messages from the client-side and
+   * emitting messages from the given Source.
+   *
+   * If you want to build the handler Flow from an independent Source and Sink, consider using one of the provided
+   * overloads of this method. If you want to construct the Flow manually, consider using [[CoupledTerminationFlow]]
+   * to bind the termination of the both sides of the Flow, as well as the [[WebSocket.ignoreSink]] for the incoming
+   * side if wanting to ignore client-side messages, as that Sink also properly handles streaming messages.
+   *
+   * Returns a response that can be used to answer a WebSocket handshake request. The connection will afterwards
+   * use the given handlerFlow to handle WebSocket messages from the client.
+   */
+  def handleMessagesWithSource(source: Graph[SourceShape[Message], _ <: Any])(implicit mat: Materializer): HttpResponse
+
+  /**
+   * Convenience API for (safely, including Streamed ones) ignoring all incoming messages from the client-side and
```

Comment:
That doesn't seem correct. Isn't that a handler that never sends out any messages?